### DISCUSSION TITLE
Cannot consume API from the browser (CORS)

### DISCUSSION THREAD

**Discussion by vkatsar** [WeblateOrg/weblate#6665] [Q&A] [answered]
Hello everybody,

I am trying to use the /api from a nuxt (vuejs) application with out success. 
I have install the docker version of weblate.

I am getting the following error.
// Request
fetch("http://localhost/api/translations/engagement-screen/engagenent-screen-component/en/file/", {
  "headers": {
    "accept": "application/json, text/plain, */*",
    "authorization": "Token UIXrULKvOuSbALf4bzndjVeAct1aHOtkjDrHqj",
    "sec-ch-ua": "\"Chromium\";v=\"94\", \"Google Chrome\";v=\"94\", \";Not A Brand\";v=\"99\"",
    "sec-ch-ua-mobile": "?0",
    "sec-ch-ua-platform": "\"macOS\""
  },
  "referrer": "http://localhost:3002/",
  "referrerPolicy": "strict-origin-when-cross-origin",
  "body": null,
  "method": "GET",
  "mode": "cors",
  "credentials": "include"
});

// Response
localhost/:1 Access to XMLHttpRequest at 'http://localhost/api/translations/engagement-screen/engagenent-screen-component/en/file/' from origin 'http://localhost:3002' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

I can't find how to configure the CORS policy through the environment variables of the docker-compose.override.yml

**Comment by nijel** [comment, marked as answer]
There is currently no way to configure this.

**Reply by vkatsar** [reply]
So the only way to use the API from the front-end is to proxy the requests? Is this going to be implemented soon or it's out of scope?

**Reply by nijel** [reply]
Patches are welcome :-).

**Comment by truesteps** [comment]
Any update on this issue? Any workaround maybe? I need this mostly for local development, on production the files will be distributed onto a CDN

**Reply by nijel** [reply]
The easiest way is to use https://pypi.org/project/django-cors-headers/:

```python
MIDDLEWARE = [
    ...
    "corsheaders.middleware.CorsMiddleware",
    ...
]

INSTALLED_APPS = [
    ...
    "corsheaders",
]


CORS_ALLOWED_ORIGINS = [
    "https://example.com",
]
CORS_URLS_REGEX = r"^/api/.*$"
```

**Reply by truesteps** [reply]
Thanks! I'll look into it

**Reply by zisuu** [reply]
I would like to apply this config, however I struggle with understanding how to achieve this: what is the easiest way to inject this config in an existing productive dockerized weblate instance without breaking anything? What happens if I add a settings-override.py in /app/data/ with only those settings? Do I need to have all installed apps and middleware present, or can I just add additions like this? Where can I see the currently loaded Middleware's and Apps if I need to include the existing ones?

**Reply by nijel** [reply]
I think we can integrate this natively, reopening https://github.com/WeblateOrg/weblate/issues/3028 for hat.
